### PR TITLE
[stable/atlantis] Support additional environment variables from kubernertes secrets

### PR DESCRIPTION
**Pull request by kpurdon**
Signed-off-by: Kyle Purdon <kylepurdon@gmail.com>

#### What this PR does / why we need it:

Adds the ability to set additional environment variables from a kubernetes secret config. 

https://kubernetes.io/docs/tasks/inject-data-application/distribute-credentials-secure/#define-container-environment-variables-using-secret-data

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

**Comment by kpurdon** [comment]
/assign @anubhavmishra

I was able to test this in my own deployment and it works as expected.

**Comment by kpurdon** [comment]
Any update on the status of this? Thanks!

**Comment by kpurdon** [comment]
@anubhavmishra @lkysow is there anything I can do to help this along? Thanks!

**Comment by lkysow** [comment]
Hey Kyle, super sorry for not reviewing this. Can you make it `environmentSecrets` and then I think it's good to go. I know the other variable is `environment` but environmentSecret just sounds weird to me.

File: stable/atlantis/README.md [hunk 1]
```diff
@@ -34,6 +34,7 @@ The following options are supported.  See [values.yaml](values.yaml) for more de
...
 | `bitbucket.baseURL`                         | Base URL of Bitbucket Server installation.                                                                                                                                                                                                                                                                | n/a     |
 | `environment`                               | Map of environment variables for the container.                                                                                                                                                                                                                                                           | `{}`    |
+| `environmentSecret`                         | Map of environment variables from kubernetes secrets for the container.                                                                                                                                                                                                                                                           | `{}`    |
```
Comment (lkysow) [inline stable/atlantis/README.md]:
```suggestion
| `environmentSecrets`                         | Array of Kubernetes secrets that can be used to set environment variables. See `values.yaml` for example.                                                                                                                                                                                                                                                          | `{}`    |
```

File: stable/atlantis/values.yaml [hunk 2]
```diff
@@ -194,3 +194,11 @@ serviceAccount:
   name:
 
 # tlsSecretName: tls
+
+# optionally specify additional environment variables to be populated from a kubernetes secret file, useful for passing in TF_VAR_foo or other secret environment variables from kubernetes secrets
+environmentSecret: {}
```
Comment (lkysow) [inline stable/atlantis/values.yaml]:
```suggestion
environmentSecrets: []
```

File: stable/atlantis/values.yaml [hunk 3]
```diff
@@ -194,3 +194,11 @@ serviceAccount:
   name:
 
 # tlsSecretName: tls
+
+# optionally specify additional environment variables to be populated from a kubernetes secret file, useful for passing in TF_VAR_foo or other secret environment variables from kubernetes secrets
```
Comment (lkysow) [inline stable/atlantis/values.yaml]:
```suggestion
# Optionally specify additional environment variables to be populated from Kubernetes secrets.
# Useful for passing in TF_VAR_foo or other secret environment variables from Kubernetes secrets.
```

**Review by lkysow — CHANGES_REQUESTED** [review]

**Comment by kpurdon** [comment]
Thanks @lkysow, applied changes and squashed. We should be good to go now!

File: stable/atlantis/Chart.yaml [hunk 4]
```diff
@@ -2,7 +2,7 @@ apiVersion: v1
 appVersion: "v0.8.2"
 description: A Helm chart for Atlantis https://www.runatlantis.io
 name: atlantis
-version: 3.6.2
+version: 3.7.2
```
Comment (jeff-knurek) [inline stable/atlantis/Chart.yaml]: when bumping a minor version, the patch version should be reset to 0
so in this case: `3.7.0`

----
https://semver.org/#spec-item-7
the same would be for major version, so if this were a PR with major changes, then it'd be `4.0.0`

**Review by jeff-knurek — COMMENTED** [review]
sorry, I also didn't review this earlier, but I have one comment that should be addressed

File: stable/atlantis/Chart.yaml (hunk 4 above)
Comment (lkysow) [reply stable/atlantis/Chart.yaml]: nice catch 🍺

**Comment by kpurdon** [comment]
@jeff-knurek @lkysow, all good now. Thanks for review.

**Comment by lukwargin** [comment]
@kpurdon  I just give it a try
```
environmentSecrets:
   - name: ENV_NAME
     secretKeyRef:
       name: secret-name
       value: secret-key
```
deployment starting but pod is not creating, I'm getting:
```
Events:
  Type     Reason        Age                   From                    Message
  ----     ------        ----                  ----                    -------
  Warning  FailedCreate  20s (x22 over 3m45s)  statefulset-controller  create Pod atlantis-0 in StatefulSet atlantis failed error: Pod "atlantis-0" is invalid: spec.containers[0].env[0].valueFrom.secretKeyRef.key: Required value
```

**Comment by lkysow** [comment]
you need key instead of value

**Pull request merged**